File: Flutter/App/lib/data/product/model/product_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product_dto.freezed.dart';
part 'product_dto.g.dart';

@freezed
class ProductDTO with _$ProductDTO {
  const factory ProductDTO({
    required int id,
    required String title,
    required String description,
    required double price,
    required String category,
    required List<String> images,
  }) = _ProductDTO;

  factory ProductDTO.fromJson(Map<String, dynamic> json) => _$ProductDTOFromJson(json);
}

File: iOS/App/ProductsCrud/Data/Product/Model/ProductDTO.swift
//
//  ProductDTO.swift
//  ProductsCrud
//
//  Created by Emil Falkowski on 11/11/2024.
//

struct ProductDTO: Decodable, Identifiable {
    let id: Int
    let title: String
    let description: String
    let price: Double
    let category: String
    let images: [String]
}
